sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -251,6 +251,30 @@ def predict(self, X):
         return np.zeros(X.shape[0])
 
 
+class NotInvariantSampleOrder(BaseEstimator):
+    def fit(self, X, y):
+        # Convert data
+        X, y = self._validate_data(
+            X, y,
+            accept_sparse=("csr", "csc"),
+            multi_output=True,
+            y_numeric=True)
+
+        self.first = X[0]
+        return self
+
+    def predict(self, X):
+        X = check_array(X)
+        # check if the sample order is different from the original
+        # or not. If the sample order is different, then just return
+        # an array of zeros.
+        # TODO: For now, im just checking if the first instance in the sample
```

Comment:
Nice use of TODO! I don't know if you meant you're still working on this (which is fine!) OR that you want to submit in the current state. If the latter was your intention, I'd encourage you instead to address this TODO first.